components/post_view/post_image/post_image.tsx

```diff
@@ -1,29 +1,30 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 
 import ExternalImage from 'components/external_image';
 import SizeAwareImage from 'components/size_aware_image';
 import ViewImageModal from 'components/view_image';
 
-export default class PostImage extends React.PureComponent {
-    static propTypes = {
-        imageMetadata: PropTypes.object.isRequired,
-        link: PropTypes.string.isRequired,
-        post: PropTypes.object.isRequired,
-    }
+type Props = {
+    imageMetadata: {
+        format: string
+    },
+    link: string,
+    post: object,
+}
 
-    constructor(props) {
+export default class PostImage extends React.PureComponent<Props> {
```

Comment:
You are missing types for state, see [here](https://github.com/mattermost/mattermost-webapp/blob/c2c57d857e5b961b7a5a32f19c016c7a88ab6f37/components/accordion.tsx#L17) for an example